#' @title Visualisations for genes.
#'
#' @description Function \code{plot_gene} generates a dashboard with methylation_path for methylation and boxplots for groups.
#'
#' @param condition.e condition for  expression
#' @param condition.m condition for methylation
#' @param data.m data for methylation
#' @param data.e data for expression
#' @param gene gene name
#'
#' @return Object of class ggplot containing visualisation of methylation on gene and boxplot for values from choosen gene broken down to groups.
#'
#'@importFrom gridExtra tableGrob
#'@importFrom grid textGrob
#'@importFrom grid gpar
#'@importFrom gridExtra grid.arrange
#'@importFrom ggplot2 ggplot_gtable
#'@importFrom ggplot2 ggplot_build
#'
#'@seealso \code{\link{plot_methylation_path}, \code{\link{plot_diff_boxplot}
#'
#'@examples
#'\dontrun{
#'condition_exp <- ifelse(BRCA_mRNAseq_chr17$SUBTYPE=="LumA","LumA","other")
#'condition_met <- ifelse(BRCA_methylation_chr17$SUBTYPE=="LumA","LumA","other")
#'plot_gene(condition_exp, condition_met, BRCA_mRNAseq_chr17, BRCA_methylation_chr17, "ICAM2")
#'}
#' @export

plot_gene <- function(condition.e, condition.m, data.e, data.m, gene){
  title <- textGrob(gene, gp=gpar(fontsize = 25), just="left")
  g <- plot_methylation_path(data.m, condition.m, gene, observ = TRUE, show_gene = TRUE) +theme(legend.position = "none")
  b1 <- plot_diff_boxplot(data.e, gene, condition.e, sqrt.trans=TRUE, title=FALSE) +theme(legend.position = c(1,1), legend.justification=c(1,1))

  tmp <- ggplot_gtable(ggplot_build(b1))
  leg <- which(sapply(tmp$grobs, function(x) x$name) ==  "guide-box")
  legend <- tmp$grobs[[leg]]


  grid.arrange(title,legend,g,b1+ theme(legend.position = "none"),heights=unit(c(20,100),"mm"),
               layout_matrix =rbind(c(1 ,1 ,2 ,2),
                                    c(3 ,3 ,3 ,4)))

}

