---
title: "Generating_reports"
output:
  pdf_document: default
  html_document: default
---

```{r, message=FALSE, warning=FALSE}
library(MLExpResso)
library(MLExpRessodata)
library(plyr)
```

Blah Blah datasets descriptions

```{r}
condition.m <-ifelse(BRCA_methylation_chr17$SUBTYPE=="LumA","LumA","other")
condition.e <-ifelse(BRCA_mRNAseq_chr17$SUBTYPE=="LumA","LumA","other")

BRCA_methylation_chr17_gene <- aggregate_probes(BRCA_methylation_chr17)
test_met_brca <- calculate_test(BRCA_methylation_chr17_gene,  condition.m, "ttest")
test_expr_brca <- calculate_test(BRCA_mRNAseq_chr17[,-1],  condition.e, "lrt")


```

```{r}


expression_100 <-test_expr_brca[order(test_expr_brca$pval, decreasing= F),]
expression_100 <- expression_100[c(1:100),]


methylation_100 <-test_met_brca[order(test_met_brca$pval, decreasing= F),]
methylation_100 <- methylation_100[c(1:100),]

x <- expression_100[which(expression_100$id %in% methylation_100$id),]

genes <- x$id


```

Function `report_generate` for generating reports:

```{r, message=FALSE, warning=FALSE, eval=FALSE}
report_generate(BRCA_mRNAseq_chr17[,-1], BRCA_methylation_chr17[,-1], condition.e, condition.m,
                test_expr_brca, test_met_brca, genes)


```

```{r, message=FALSE, warning=FALSE}
plot_gene(BRCA_methylation_chr17[,-1], BRCA_mRNAseq_chr17[,-1],condition.m, condition.e,"CBX2")
```

```{r, message=FALSE, warning=FALSE}
plot_volcanoes(BRCA_methylation_chr17[,-1], BRCA_mRNAseq_chr17[,-1], condition.m, condition.e , "CBX2", list(test_expr_brca), list(test_met_brca), values=TRUE)

```
