---
title: "MLExpResso: differential expression and methylation analysis"
subtitle: "Case study using RTCGA data"
author: "Aleksandra DÄ…browska, Alicja Gosiewska"
output: 
  pdf_document:
    number_sections: true
toc: TRUE
vignette: >
  %\VignetteIndexEntry{MLExpResso: differential expression and methylation analysis}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

# Introduction
The following guide presents the effect of the MLExpResso package on the analysis of changes in expression and methylation of the human genome.

Scientists believe that the result of increased methylation is decreased gene expression. Recent studies suggest that the relationship between methylation and expression is more complex than we previously thought.

`MLExpResso` is an R package for integrative analyses and visualization of gene expression and DNA methylation data.

Key functions of this package are:

- Identification of genes with affected expression,

- Identification of DMR - differentially methylated regions,

- Identification of regions with changes in expression and methylation,

- Visualization of identified regions.

For both, methylation and expression data, we conduct some statistical tests and present graphically received results.
The joint modeling and visualization of genes expression and methylation improve interpretability of identified signals. 

The methodology is supplemented with example applications to The Cancer Genome Atlas data.

# Standard Workflow

In this vignette we will work with the data sets containing information about gene expression and methylation
for patients with breast cancer. We will analyze differences in methylation and expression for patients with
different subtypes of BRCA cancer. To run the examples below you should install `MLExpRessoData` package
(https://github.com/geneticsMiNIng/MLGenSigdata). Data sets in this R package are based on the Bioconductor
package `RTCGA` (https://bioconductor.org/packages/release/bioc/html/RTCGA.html).

```{r MLExpResso, message=FALSE, warning=FALSE}
library(MLExpResso)
library(MLExpRessoData)
```

## Identification of genes with affected expression

### `BRCA_exp`
Package `MLExpRessoData` contains `BRCA_exp` dataset. This set contains information about gene expression:
read counts per-gene, computed for genes for 736 patients with breast cancer. Rows of this data set correspond
to samples taken from patients. First column `SUBTYPE` corresponds to a subtype of BRCA cancer, next
columns correspond to genes.

```{r}
BRCA_exp[1:5, 1:5]
```

Before we go to the testing, we need to define condition values for each sample. We would like to test for differences between `LumA` subtype and `other` subtypes of breast cancer, so we create a vector, which each element corresponds to a sample. Our division into this two groups relies on numbers of occurences of each subtype. The `LumA` subtype is the most common, in case of breast cancer.

```{r, message=FALSE}
condition_exp <- ifelse(BRCA_exp$SUBTYPE == "LumA", "LumA", "other")
head(condition_exp, 8)
```

Now we can visualize mean expression of each gene in a division of groups Luma and other.
To do this we use the `plot_means()` function.

This function requires parameters such that:

* `data` - data set with information from methylation or expression. In the example below, we use the `BRCA_exp` data without the `SUBTYPE` column. 

* `condition` - a factor of levels corresponding to order of samples in data. In our example, we use the `condition_exp` vector defined earlier.

* `names` - number of genes to be labeled. On a plot are marked genes with the smallest geometric mean of means.

```{r}
plot_means(
  data = BRCA_exp[,!(colnames(BRCA_exp) == "SUBTYPE")], 
  condition = condition_exp, 
  names = 5
)
```


### Testing
In the `MLExpResso` package we carry out the tests for identification of genes with affected expression. To do
this we use the `calculate_test()` function. 

This function requires parameters such that:


* `data` - object of the class appropriate for the given test. In the example below, we use the `BRCA_exp` data without the `SUBTYPE` column. 

* `condition` - the factor of levels corresponding to order of samples in data. In our example, we use the `condition_exp` vector defined earlier.

* `test` - character defining test. Possible values of parameter `test` are described in the documentation of this function.

Tests are based on the methods implemented in Bioconductor packages `Deseq`, `Deseq2` and `edgeR`.

```{r}
res_exp <- calculate_test(
  data = BRCA_exp[,!(colnames(BRCA_exp) == "SUBTYPE")], 
  condition = condition_exp, 
  test = "lrt"
)
head(res_exp)
```

As a result we get a data frame with columns describing the log2 of the fold change, p-value and the mean for each gene.


## Identification of DMR - differentially methylated regions

### `BRCA_met` data set
In this section, we will work with the methylation level data from TCGA database. Package `MLExpRessoData`
contains `BRCA_met` dataset. This data set contains information about methylation of CpG probes for patients
with breast cancer. Rows of this data set correspond to patients, more precisely, to samples taken from
patients. First column `SUBTYPE` corresponds to a subtype of BRCA cancer, next columns correspond to CpG
probes. Values inside the table indicate the percentage methylation level of CpG probe for specified sample.

```{r}
head(BRCA_met)[1:5, 1:4]
```

###Data preparation
In this analysis we would like to find genes with different methylation. At first we need to use
function `aggregate_probes()`, which generates new data frame with CpG probes aggregated to
genes. To this aggregation we use, by default, the Illumina Human Methylation data set from the
`TxDb.Hsapiens.UCSC.hg18.knownGene` Bioconductor package.

Function `aggregate_probes()` requires a parameter `data` -  data frame containing methylation values for CpG probes.

```{r mapping}
BRCA_met_gen <- aggregate_probes(data = BRCA_met) 
head(BRCA_met_gen)[1:5, 1:4]
```


In our example we will test for differential methylation between groups with `LumA` breast cancer subtype and
other subtypes of that cancer. Again we will use condition vector, which consist of two values corresponds
to a subtype of breast cancer: `LumA` and `other`.

```{r}
condition_met <- ifelse(BRCA_met$SUBTYPE == "LumA", "LumA", "other")
head(condition_met, 8)
```

### Testing

In the `MLExpResso` package we carry out the tests for identification of differentially methylated regions.
To do this we use the `calculate_test()` function.

This function requires parameters such that:


* `data` - object of the class appropriate for the given test. In the example below, we use the `BRCA_met_gen` methylation data aggregated to probes. 

* `condition` - a factor of levels corresponding to order of samples in data. In our example, we use the `condition_met` vector defined earlier.

* `test` - character defining test. Possible values of parameter `test` are described in the documentation of this function. 

Methylation tests are based on the methods implemented in packages `limma` and `MethyAnalysis`.

```{r}
res_met <- calculate_test(
  data = BRCA_met_gen, 
  condition = condition_met,
  test = "ttest"
)
head(res_met)
```

As a result we get a data frame with columns describing the log2 of the fold change, p-value and the mean for each gene.

## Identification of regions with changes in expression and methylation

We can also create a comparison table with results of `calculate_test()` function for methylation and
expression data.

Function `calculate_comparison_table()` requires parameters such that:


* `data1`, `data2` - objects of the class appropriate for the given tests.  

* `condition1`, `condition2` - factors of levels coresponding to order of samples in `data1` and `data2` respectively.

* `test1`, `test2` - characters defining tests coresponding to order of samples in `data1` and `data2` respectively. Possible values of parameter `test` are described in `calculate_test()` function documentation. 

```{r, message=FALSE, warning=FALSE}
comparison <- calculate_comparison_table(
  data1 = BRCA_exp[,!(colnames(BRCA_exp)=="SUBTYPE")], 
  data2 = BRCA_met_gen, 
  condition1 = condition_exp, 
  condition2 = condition_met,
  test1 = "nbinom2", 
  test2 = "ttest"
)
head(comparison)
```
As a result, we get a data frame with columns describing the log2 of the fold change, p-value and the mean for each gene for two tests. With this two test results, we compute the ranking of the most significant changed genes in terms of both methylation and expression. The created column contains the geometric mean of p-values for expression and methylation.

We can visualize results of `calculate_comparison_table()` by `plot_pvalues()` function.
This function requires parameters such that:

* `data` - data.frame - result of `calculate_comparison_table()` function.   

* `names` - number of genes to be labeled. Gens are selected based on the ranking of the most significant changed genes in terms of both methylation and expression - geom.mean.rank column. 


```{r}
plot_pvalues(comparison, names = 10)
```

## Visualization of identified regions

The great advantage of `MLExpResso` package is the ability to perform a variety of visualizations for expression
and methylation.
All plots in our package are based on the `ggplot2` package. We use also the `scales` and `ggrepel` packages
for mathematical axes and repel overlapping text labels. 

```{r}
test_exp <- comparison[ , c("id", "nbinom2.log2.fold", "nbinom2.pval")]
test_met <- comparison[ , c("id", "ttest.log2.fold", "ttest.pval")]
```
  
  
For both, methylation and expression data, we can visualise the volcano plots for results of chosen tests and simple statistics for chosen gene. 

Function `plot_volcanoes()` requiers parameters such that:

* `data.m`, `data.e` - data sets with information from methylation and expression respectively. In the example below, we use the `BRCA_met` and `BRCA_exp` data without the `SUBTYPE` columns.

* `condition.m`, `condition.e` -  factors of levels coresponding to the order of samples in `data.m` and `data.e` respectively. In our example, we use the `condition_met` and `condition_exp` vectors defined earlier.

* `gene` - character defining the gene name. In the example, we visualize results for `CACNA1G` gene

* `test.m`, `test.e` - results from `calculate_test()` function for methylation and expression data.

* `values` - if `TRUE` p-values and log fold for the chosen gene will be added to the plot.

```{r, message=FALSE, warning=FALSE, fig.width=19, fig.height=10}
plot_volcanoes(
  data.m = BRCA_met[,!(colnames(BRCA_met) == "SUBTYPE")],
  data.e = BRCA_exp[,!(colnames(BRCA_exp) == "SUBTYPE")],
  condition.m = condition_met, 
  condition.e = condition_exp, 
  gene = "CACNA1G", 
  test.m = test_met, 
  test.e = test_exp,
  values=TRUE
)
```

Other function `plot_gene()` allow us to visualize the `methylation path` - placement of probes near the
gene with a marked percentage of methylation for each probe in division into groups. 

Function `plot_gene()` requiers parameters such that:

* `data.m`, `data.e` - data sets with informations from methylation and expression respectively. Note that `plot_gene()` methylation requires data frame with CpG probes, not genes. In the example below, we use the `BRCA_met` and `BRCA_exp`.

* `condition.m`, `condition.e` -  factors of levels coresponding to order of samples in `data.m` and `data.e` respectively. In our example, we use the `condition_met` and `condition_exp` vectors defined earlier.

* `gene` - character defining the gene name. In the example, we visualize results for `CACNA1G` gene.

* ... additional parameters. Below we use parameters `show.gene` - If `TRUE` line corresponding to the gene will be drawn, `observ` - If `TRUE` dots corresponding to CpG probes will be drawn and `islands` -If `TRUE` line corresponding to islands should be drawn.

```{r, message=FALSE, warning=FALSE, fig.width=19, fig.height=5.5}
plot_gene(
  data.m = BRCA_met, 
  data.e = BRCA_exp, 
  condition.m = condition_met, 
  condition.e = condition_exp, 
  gene = "CACNA1G",
  show.gene = TRUE,
  observ = TRUE,
  islands = TRUE
)
```

Using this function we also get boxplots containing values of expression in division from `condition_exp` vector for chosen gene.
