---
title: "Usage - survival status case"
author: "Aleksandra Dąbrowska, Alicja Gosiewska"
date: '`r Sys.Date()`'
output:
  pdf_document: default
  html_document: default
toc: yes
vignette: |
  %\VignetteIndexEntry{Vignette Title} %\VignetteEngine{knitr::rmarkdown} %\VignetteEncoding{UTF-8}
---

# Standard Workflow

In this vignette we will work with the data sets containing information about gene expression and methylation for patients with breast cancer. We will analyze differences in methylation and expression for patients with different subtypes of BRCA cancer.


## Function test_diff

The main function of the package is `test_diff`. It allows to find differences between genes methylation or expression, taking into account additional information about samples.

### Methylation

Methylation is a process by which methyl groups are added to the DNA molecule. It can change the activity of a DNA without changing the sequence. DNA methylation typically acts to repress gene transcription. But there exist situations in which adding the methyl groups intensify it.
DNA methylation is associated with a lots of key processes including genomic imprinting, repression of transposable elements, aging and carcinogenesis.
In our work we want to bind methylation process and carcinogenesis.

#### `BRCA_methylation_all_surv` data set
In this section, we will work with the methylation level data from TCGA database. 
Package contains `BRCA_methylation_all_surv` dataset. `BRCA_methylation_all_surv` contains information about methylation of CpG islands for patienst with breast cancer. 
Rows of this data set correspond to patients, more precisely, to samples taken from patients.
First column `SUBTYPE`corresponds to a subtype of BRCA cancer, next column to a survival status, more precisely: `1` corresponds to `Dead`, `0` to `Alive`.
We divided this column in the following way:
-patients who have observation time longer than 3 years and any vital status we assign to `0` group
-patients who have observation time shorter than 3 years and `Dead` in vital status we assign to `1` group
-we disregarded patients not belonging to previous groups.

Other columns correspond to CpG islands.
Values inside the table indicate the methylation level of CpG island for specified sample.
```{r}
library(MetExpR)
head(BRCA_methylation_all_surv)[1:5,1:4]
```

#### Data preparation

In this analysis we would like to find genes with different methylation level. At first we need to use function `map_to_gene`, which generates new data frame with CpG islands mapped to genes.

```{r mapping}
BRCA_methylation_gen <- map_to_gene(BRCA_methylation_all_surv[,-c(1,2)]) 
head(BRCA_methylation_gen[,-1])[1:5,1:4]
#cos nie tak w mappowaniu(wyrzuca pierwszą kolumne) - poprawic
```

Function `test_diff` allows us to test for differences between the base means for two or more conditions. 

In this case we have two conditions, connected with survival status.


```{r}
condition <- ifelse(BRCA_methylation_all_surv$survival_status==1, "Dead","Alive")
#zera i jedynki nie sa dobrym pomyslem-
#dostajemy error przy wywolaniu makeContrasts 
#Error in makeContrasts(contrasts = forms, levels = design) : 
#  The levels must by syntactically valid names in R, see help(make.names).  Non-valid names: 0,1
```

```{r, include=FALSE}
#dla szybszych obliczen - wezmiemy jakies 100 genow
x <- colnames(BRCA_methylation_all_surv[,-c(1,2)])
wyspy <- sample(x,200)
xxx <- BRCA_methylation_all_surv[,which(colnames(BRCA_methylation_all)%in% wyspy)]
BRCA_methylation_gen <- map_to_gene(xxx)
BRCA_methylation_gen <- BRCA_methylation_gen[,-1]
BRCA_methylation_gen <- cbind(BRCA_methylation_all_surv[,c(1,2)],BRCA_methylation_gen)
BRCA_methylation_all_surv <- cbind(BRCA_methylation_all_surv,xxx)

geny <- colnames(BRCA_methylation_gen)

gen <- geny[10]


condition <- ifelse(BRCA_methylation_all_surv$survival_status==1, "Dead","Alive")
```

#### T-test

One of the most used tools for testing differences between values is t-test. The null hypothesis we have consider, is that means in two groups are equal.
To use it in `test_diff` function, we set value of parameter `test` on "ttest".

```{r}
test.mety <- test_diff(BRCA_methylation_gen[,-c(1,2)], condition, test="ttest")
```

As a result we obtain a data frame with columns corresponds to: id of gene, mean, logarithm of fold change, p-value for t-test, adjusted p-value (BH method). For more information about customizing this function see the help page for `test_diff`.

```{r}
head(test.mety)
```

### Expression 

Gene expression is the process by which information from a gene is used in the synthesis of proteins. The process of gene expression is used by all known life.

#### `BRCA_mRNAseq_all_surv` data set

In this section we will use data set `BRCA_mRNAseq_all_surv`, which contains information about gene expression.
Rows of this data set correspond to samples taken from patients.
First column `SUBTYPE`corresponds to a subtype of BRCA cancer, next column, like in `BRCA_methylation_all_surv` to the survival status, next columns correspond to genes.

```{r}
BRCA_mRNAseq_all_surv[1:5,1:5]
```

#### Negative binomial test
Negative binomial test, which uses negative binomial distribution is an another tool for finding differencial expression between our conditions. 

As in the t-test we also need a description of the samples, which we keep in a vector, whose elements correspond to different gorups.

In our example we will test for differential expression between groups with LumA breast cancer subtype and other subtypes of that cancer.
Again we will use vector `conditions`, which consist of two values corresponds to subtype of breast cancer: LumA and other.
```{r, include=FALSE}
BRCA_mRNAseq_all_surv <- BRCA_mRNAseq_all_surv[,which(colnames(BRCA_mRNAseq_all_surv)%in%geny)]
```

```{r, message=FALSE}
condition<-ifelse(BRCA_mRNAseq_all_surv$survival_status==1,"Dead","Alive")
head(condition,8)
```

For using negative binomial test, in function `test_diff` we set value "nbinom2" for parameter `test`. (negative binomial test from DESeq2 package)
```{r}
test.expr <- test_diff(BRCA_mRNAseq_all_surv[,-c(1,2)], condition, test="nbinom2")
```

As a result we obtain the following data frame:

```{r}
head(test.expr)
```

# Visualization

## em_plot
```{r}
em_plot(BRCA_mRNAseq_all_surv[,-c(1,2)], condition, names=5)
```

## log-log p-value

Firstly, we want to visualise the p-values for expression and methylation from negative binomial test and t-test respectively.

```{r p_values_plot}
p_values_plot(test.expr, test.mety)
```

Additionally, `names` parameter allows to mark genes with sum of p-values for methylation and expression, lower than given value. Value of parameter `names` defines, number of genes to label.
```{r p_values_plot2}
p_values_plot(test.expr, test.mety, names = 5)
``` 

To read more about `p_values_plot` (e.g other ways to labeling genes) see help page for that function.

## Volcano plot

For identify changes in our data sets we use a volcano plot - some type of scatter-plot.
It plots logarithm of p-value versus logarithm of fold-change on the y and x axes, respectively. 

```{r}
volcano_plot(test.expr)
```

Function `volcano_plot` has parameters that allow to better analyze the results: `line` and `names`. The `line` parameter allows to set the horizontal line on plot on selected value. The `names` parameter signs choosen number of genes with the lowest p-value.

```{r}
volcano_plot(test.expr, line = 0.05)
volcano_plot(test.expr, names = 10)
```

## Methylation and expression for one gene.

In the end we want to present the distribution of methylation and expression for choosen genes.

```{r, include=FALSE}
# Warning in install.packages :
#   package ‘easyGgplot2’ is not available (for R version 3.3.3)

#library(easyGgplot2)
#p1 <-genereg_vs_met(BRCA_methylation_all_surv, "ACPT")
#p2 <-boxplot_gene_expr(BRCA_mRNAseq_all_surv,"ACPT")
#ggplot2.multiplot(p1,p2)

```

Function `CpG_mean` computes methylation means of CpG islands for choosen gene. In this case: "BRCA1"

```{r}
BRCA1_gene <- CpG_mean(BRCA_methylation_all_surv, gen)
BRCA1_gene
```


###Methylation and expression in groups.

Two subtype groups of cancer in one plot.
```{r}
condition <- ifelse(BRCA_methylation_all_surv$survival_status==1,"Dead", "Alive")
genereg_vs_met(BRCA_methylation_all_surv, condition, gen)

genereg_vs_met(BRCA_methylation_all_surv, condition, gen, show_gen=TRUE)
```

```{r, include=FALSE}
library(gridExtra)
library(grid)
library(ggplot2)
library(edgeR)
condition.e <- ifelse(BRCA_mRNAseq_all_surv$survival_status==1,"Dead","Alive")
condition.m <- ifelse(BRCA_methylation_all_surv$survival_status==1,"Dead", "Alive")
g <- genereg_vs_met(BRCA_methylation_all_surv, condition.m, gen, observ = TRUE, show_gen = TRUE)
#normalization * 10000
BRCA_mRNAseq_all_surv_cpm <- cpm(BRCA_mRNAseq_all_surv[,-c(1,2)])
b1 <- boxplot_that(BRCA_mRNAseq_all_surv_cpm,gen,condition.e)
v.e <- volcano_plot(test.expr, ngen =gen,ylog=TRUE, title="expression", fold_line = 2, line=0.05)


v.m <- volcano_plot(test.mety,ngen = gen,ylog=TRUE,title="methylation",line=0.05)

s <- tableGrob(t(sum_gen(BRCA_mRNAseq_all_surv[,-c(1,2)],condition.e , gen)))

title <- textGrob("Summary", vjust = 5, gp=gpar(fontsize=20))

grid.arrange(g,b1,v.m, v.e,  
             layout_matrix =rbind(c(1 ,1 ,1 ,NA),
                                  c(1 ,1 ,1 ,2 ),
                                  c(1 ,1 ,1 ,2 ),
                                  c(3 ,3 ,4 ,4 ),
                                  c(3 ,3 ,4 ,4 ),
                                  c(3 ,3 ,4 ,4 )))





```



```{r, message=FALSE, fig.height=10, fig.width=20}
visual_gene(condition.e, condition.m, BRCA_methylation_all_surv[,-c(1,2)],BRCA_mRNAseq_all_surv[,-c(1,2)], gen, test.expr, test.mety)


visual_volc(condition.e, condition.m, BRCA_methylation_all_surv[,-c(1,2)],BRCA_mRNAseq_all_surv[,-c(1,2)], gen, list(nbinom =test.expr), list(ttest=test.mety))
```

```{r, message=FALSE, fig.width=19, fig.height=20}
visual_volc(condition.e, condition.m, BRCA_methylation_all_surv[,-c(1,2)],BRCA_mRNAseq_all_surv[,-c(1,2)], gen, list(nbinom=test.expr,nbinom2=test.expr), list(ttest=test.mety,test.mety))


```