---
title: "Vignette Title"
author: "Aleksandra DÄ…browska, Alicja Gosiewska"
date: "`r Sys.Date()`"
toc: yes
output: pdf_document
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Package
Package ...   the detection of differentially expressed and methylated genes. 


# Standard Workflow

In this vignette we will work with the data sets containing information about gene expression and methylation for patients with breast cancer. We will analyze differences between methylation and expression for patients with different subtypes of BRCA cancer.

## Function test_diff

The main function of the package is `test_diff`. It allows to find differences between genes methylation or expression, taking into account additional information about samples.

### Methylation
Info about methylation...

In this section, we will work with the methylation level data from TCGA database.  
Data set `BRCA_methylation_chr17` contains information about methylation of CpG islands located on 17th chromosome for patienst with breast cancer.
```{r}
load("BRCA_methylation_chr17.rda")
head(BRCA_methylation_chr17)[1:5,1:4]
```

In this analysis we would like to find genes with different methylation. At first we need to use function `map_to_gene`, which generates new data frame with CpG islands mapped to genes.

```{r}
library(MetExpR)
BRCA_methylation_chr17_gen <- map_to_gene(BRCA_methylation_chr17[,-1]) 
head(BRCA_methylation_chr17_gen)[1:5,1:4]
```

Function `test_diff` allows us to test for differences between the base means for two or more conditions. 

Before we go to the testing, we need to define condition values for each sample. We would like to test for differences between LumA subtype and other subtypes of breast cancer, so we create vector, which each element corresponds to a sample.

```{r}
condition <- ifelse(BRCA_methylation_chr17$SUBTYPE=="LumA","LumA", "other")
head(condition,8)
```

#### T-test

One of the tools for testing differences between values is t-test. To use it in `test_diff` function, we set value of parameter `test` on "ttest".
```{r}
test.mety <- test_diff(BRCA_methylation_chr17_gen, condition, test="ttest")
```

As a result we obtain a data frame with columns corresponds to: id of gene, mean, logarithm of fold change, p-value for t-test, adjusted p-value (BH method). For more information about customizing this function see the help page for `test_diff`.

```{r}
head(test.mety)
```

### Expression
Info about expression...

In this section we will use data set BRCA_mRNAseq_chr17, which contains information about gene expression.
This data set contains per-gene read counts computed for genes from 17th chromosome for 100 patients with breast cancer.
```{r}
load("BRCA_mRNAseq_chr17.rda")
```


```{r}
BRCA_mRNAseq_chr17[1:5,1:5]
```

#### Nbinom test
Negative binomial test is an another tool for finding differences between the base means of data having two or more conditions. 

As in the t-test w need also a description of the samples, which we keep in a vector, whose elements correspond to different gorups.

In our example we will test for differential expression between groups with LumA breast cancer subtype and other subtypes of that cancer.
Again we will use vector `conditions`, which consist of two values corresponds to subtype of breast cancer: LumA and other.
```{r, message=FALSE}
condition<-ifelse(BRCA_mRNAseq_chr17$SUBTYPE=="LumA","LumA","other")
head(condition,8)
```

```{r, echo=FALSE}
load("testexpr.rda")
```

For using negative binomial test, in function `test_diff` we set value "nbinom" for parameter `test`.
Evaluation for nbinorm may take a few minutes
```{r,eval=FALSE}
test.expr <- test_diff(BRCA_mRNAseq_chr17[,-1], condition, test="nbinom")
```

As a result we obtain the following data frame:

```{r}
head(test.expr)
```

# Visualization

## log-log p-value

```{r}
p_values_plot(test.expr, test.mety)
```

## volcano

Volcan plot for expression

```{r}
volcano_plot(test.expr, test.mety, type="expression")
```


## methyl "road" + boxplots for expr

```{r}

```



