---
title: "MLExpResso: differential expression and methylation analysis"
subtitle: "Case study using RTCGA data"
author: "Aleksandra DÄ…browska, Alicja Gosiewska"
output: 
  pdf_document:
    number_sections: true
toc: TRUE
---

# Introduction
It is considered that the result of increased methylation is decreased gene expression. While, recent studies suggest that the relationship between methylation and expression is more complex than was previously thought.

`MLExpResso` is an R package for integrative analyses and visualization of gene expression and DNA methylation data.

Key functions of this package are:

- Identification of DMR - differentially methylated regions,

- Identification of differentially expressed genes,

- Identification regions with changes in expression and methylation,

- Visualization of identified regions.

The joint modeling and visualization of genes expression and methylation improve interpretability of identified signals.

The methodology is supplemented with example applications to The Cancer Genome Atlas data.

# Standard Workflow

In this vignette we will work with the data sets containing information about gene expression and methylation for patients with breast cancer. We will analyze differences in methylation and expression for patients with different subtypes of BRCA cancer. To run the examples below you should install `MLExpRessoData` package (https://github.com/geneticsMiNIng/MLGenSigdata).
Data sets in this R package are based on Bioconductor package `RTCGA` (https://bioconductor.org/packages/release/bioc/html/RTCGA.html).

The vignette below was created using the `roxygen2` package.

```{r MLExpResso, message=FALSE, warning=FALSE}
library(MLExpResso)
library(MLExpRessoData)
```

## Differentially expressed genes

### `BRCA_exp`
Package `MLExpRessoData` contains `BRCA_exp` dataset. 
This set contains information about gene expression: read counts per-gene, computed for genes for 736 patients with breast cancer.
Rows of this data set correspond to samples taken from patients.
First column `SUBTYPE` corresponds to a subtype of BRCA cancer, next columns correspond to genes.

```{r}
BRCA_exp[1:5,1:5]
```

In our example we will test for differential expression between groups with `LumA` breast cancer subtype and `other` subtypes of that cancer.
Again we will use vector `conditions`, which consist of two values corresponds to subtype of breast cancer: `LumA` and `other`.
```{r, message=FALSE}
condition_exp <- ifelse(BRCA_exp$SUBTYPE == "LumA","LumA","other")
head(condition_exp, 8)
```

### Testing
In the `MLExpResso` package we carry out the tests for identification of genes with affected expression. To do this we use the `calculate_test()` function. Possible values of parameter `test` are described in function documentation.
Expression tests are based on the methods implemented in packages `Deseq`, `Deseq2` and `edgeR`.

```{r}
res_exp <- calculate_test(data = BRCA_exp[,!(colnames(BRCA_exp) == "SUBTYPE")], 
                          condition = condition_exp, 
                          test = "lrt")
head(res_exp)
```

## Differentially methylated regions (DMR)

### `BRCA_met` data set
In this section, we will work with the methylation level data from TCGA database. 
Package `MLExpRessoData` contains `BRCA_met` dataset. This data set contains information about methylation of CpG probes for patients with breast cancer. 
Rows of this data set correspond to patients, more precisely, to samples taken from patients.
First column `SUBTYPE` corresponds to a subtype of BRCA cancer, next columns correspond to CpG probes.
Values inside the table indicate the percentage methylation level of CpG probe for specified sample.


```{r}
head(BRCA_met)[1:5,1:4]
```

### Data preparation
In this analysis we would like to find genes with different methylation. At first we need to use function `aggregate_probes()`, which generates new data frame with CpG probes aggregated to genes. To this aggregation we use, by default, the Illumina Human Methylation data set from the `TxDb.Hsapiens.UCSC.hg18.knownGene` Bioconductor package.

```{r mapping}
BRCA_met_gen <- aggregate_probes(data = BRCA_met) 
head(BRCA_met_gen)[1:5,1:4]
```


Before we go to the testing, we need to define condition values for each sample. We would like to test for differences between `LumA` subtype and `other` subtypes of breast cancer, so we create a vector, which each element corresponds to a sample. Our division into this two groups relies on numbers of occurences of each subtype. The `LumA` subtype is the most common, in case of breast cancer.

```{r}
condition_met <- ifelse(BRCA_met$SUBTYPE == "LumA","LumA", "other")
head(condition_met, 8)
```

### Testing

In the `MLExpResso` package we carry out the tests for identification of differentially methylated regions. To do this we use the `calculate_test()` function. Possible values of parameter `test` are described in function documentation.
Methylation tests are based on the methods implemented in packages `limma` and `MethyAnalysis`.

```{r}
res_met <- calculate_test(data = BRCA_met_gen, 
                          condition = condition_met, 
                          test = "ttest")
head(res_met)
```


## Comparison of test results

We can also create a comparison table with results of `calculate_test()` function for methylation and expression data. With thid two results we compute the ranking of the most significant changed genes in terms of both methylation and expression. The created column contains the geometric mean of p-values for expression and methylation. 

```{r, message=FALSE, warning=FALSE}
genes_comparison<-calculate_comparison_table(data1 = BRCA_exp[,!(colnames(BRCA_exp) == "SUBTYPE")], 
                                               data2 = BRCA_met_gen, 
                                               cond1 = condition_exp, 
                                               cond2 = condition_met,

                                               test1 = "nbinom2", 
                                               test2 = "ttest")

head(genes_comparison)
```


## Visualization
The great advantage of `MLExpResso` package is the ability to perform a variety of visualizations for expression and methylation.

All plots in our package are based on the `ggplot2` package.  We use also the `scales` and `ggrepel` packages for mathematical axes and repel overlapping text labels.

```{r}
test_exp <- genes_comparison[ ,c("id","nbinom2.log2.fold","nbinom2.pval")]
test_met <- genes_comparison[ ,c("id","ttest.log2.fold","ttest.pval")]
```

  
For both, methylation and expression data, we can visualise the volcano plots for results of chosen tests and simple statistics for chosen gene. 
```{r, message=FALSE, warning=FALSE, fig.width=19, fig.height=10}
plot_volcanoes(data.m = BRCA_met[,!(colnames(BRCA_met) == "SUBTYPE")],
               data.e = BRCA_exp[,!(colnames(BRCA_exp) == "SUBTYPE")], 
               condition.m =  condition_met, 
               condition.e = condition_exp, 
               gene = "CACNA1G", 
               test.m = test_met, 
               test.e = test_exp, 
               values=TRUE)

```

Other function `plot_gene()` allow us to visualise the `methylation path` - placement of probes near the gene with a marked percentage of methylation for each probe in division into groups. Using this function we also get boxplots containing values from expression in division from `condition_exp` vector for chosen gene. Note that `plot_gene()` methylation require data frame with CpG probes, not genes.

```{r, message=FALSE, warning=FALSE, fig.width=19, fig.height=5.5}
plot_gene(data.m = BRCA_met, 
          data.e = BRCA_exp, 
          condition.m = condition_met, 
          condition.e = condition_exp, 
          gene = "CACNA1G")
```


