<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MLExpResso: differential expression and methylation analysis • MLExpResso</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">MLExpResso</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QuickStart.html">MLExpResso: differential expression and methylation analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>MLExpResso: differential expression and methylation analysis</h1>
            <h3 class="subtitle">Case study using RTCGA data</h3>
                        <h4 class="author">Aleksandra Dąbrowska, Alicja Gosiewska</h4>
            
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The following guide presents the effect of the MLExpResso package on the analysis of changes in expression and methylation of the human genome.</p>
<p>Scientists believe that the result of increased methylation is decreased gene expression. Recent studies suggest that the relationship between methylation and expression is more complex than we previously thought.</p>
<p><code>MLExpResso</code> is an R package for integrative analyses and visualization of gene expression and DNA methylation data.</p>
<p>Key functions of this package are:</p>
<ul>
<li><p>Identification of genes with affected expression,</p></li>
<li><p>Identification of DMR - differentially methylated regions,</p></li>
<li><p>Identification of regions with changes in expression and methylation,</p></li>
<li><p>Visualization of identified regions.</p></li>
</ul>
<p>For both, methylation and expression data, we conduct some statistical tests and present graphically received results. The joint modeling and visualization of genes expression and methylation improve interpretability of identified signals.</p>
<p>The methodology is supplemented with example applications to The Cancer Genome Atlas data.</p>
</div>
<div id="standard-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#standard-workflow" class="anchor"></a>Standard Workflow</h1>
<p>In this vignette we will work with the data sets containing information about gene expression and methylation for patients with breast cancer. We will analyze differences in methylation and expression for patients with different subtypes of BRCA cancer. To run the examples below you should install <code>MLExpRessoData</code> package (<a href="https://github.com/geneticsMiNIng/MLGenSigdata" class="uri">https://github.com/geneticsMiNIng/MLGenSigdata</a>). Data sets in this R package are based on the Bioconductor package <code>RTCGA</code> (<a href="https://bioconductor.org/packages/release/bioc/html/RTCGA.html" class="uri">https://bioconductor.org/packages/release/bioc/html/RTCGA.html</a>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(MLExpResso)
<span class="kw">library</span>(MLExpRessoData)</code></pre></div>
<div id="identification-genes-with-affected-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#identification-genes-with-affected-expression" class="anchor"></a>Identification genes with affected expression</h2>
<div id="brca_exp" class="section level3">
<h3 class="hasAnchor">
<a href="#brca_exp" class="anchor"></a><code>BRCA_exp</code>
</h3>
<p>Package <code>MLExpRessoData</code> contains <code>BRCA_exp</code> dataset. This set contains information about gene expression: read counts per-gene, computed for genes for 736 patients with breast cancer. Rows of this data set correspond to samples taken from patients. First column <code>SUBTYPE</code> corresponds to a subtype of BRCA cancer, next columns correspond to genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">BRCA_exp[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</code></pre></div>
<pre><code>##                              SUBTYPE AANAT AARSD1 AATF AATK
## TCGA-A1-A0SB-01A-11R-A144-07  Normal     9   2354 2870  317
## TCGA-A1-A0SD-01A-11R-A115-07    LumA     2   1846 5656  312
## TCGA-A1-A0SE-01A-11R-A084-07    LumA    11   3391 9522  736
## TCGA-A1-A0SF-01A-11R-A144-07    LumA     0   2169 4625  169
## TCGA-A1-A0SG-01A-11R-A144-07    LumA     1   2273 3473   92</code></pre>
<p>Before we go to the testing, we need to define condition values for each sample. We would like to test for differences between <code>LumA</code> subtype and <code>other</code> subtypes of breast cancer, so we create a vector, which each element corresponds to a sample. Our division into this two groups relies on numbers of occurences of each subtype. The <code>LumA</code> subtype is the most common, in case of breast cancer.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">condition_exp &lt;-<span class="st"> </span><span class="kw">ifelse</span>(BRCA_exp<span class="op">$</span>SUBTYPE <span class="op">==</span><span class="st"> "LumA"</span>, <span class="st">"LumA"</span>, <span class="st">"other"</span>)
<span class="kw">head</span>(condition_exp, <span class="dv">8</span>)</code></pre></div>
<pre><code>## [1] "other" "LumA"  "LumA"  "LumA"  "LumA"  "LumA"  "other" "LumA"</code></pre>
</div>
<div id="testing" class="section level3">
<h3 class="hasAnchor">
<a href="#testing" class="anchor"></a>Testing</h3>
<p>In the <code>MLExpResso</code> package we carry out the tests for identification of genes with affected expression. To do this we use the <code><a href="../reference/calculate_test.html">calculate_test()</a></code> function.</p>
<p>This function requires parameters such that:</p>
<ul>
<li><p><code>data</code> - object of the class appropriate for the given test. In the example below, we use the <code>BRCA_exp</code> data without the <code>SUBTYPE</code> column.</p></li>
<li><p><code>condition</code> - the factor of levels corresponding to order of samples in data. In our example, we use the <code>condition_exp</code> vector defined earlier.</p></li>
<li><p><code>test</code> - character defining test. Possible values of parameter <code>test</code> are described in the documentation of this function.</p></li>
</ul>
<p>Tests are based on the methods implemented in Bioconductor packages <code>Deseq</code>, <code>Deseq2</code> and <code>edgeR</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_exp &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calculate_test.html">calculate_test</a></span>(
  <span class="dt">data =</span> BRCA_exp[,<span class="op">!</span>(<span class="kw">colnames</span>(BRCA_exp) <span class="op">==</span><span class="st"> "SUBTYPE"</span>)], 
  <span class="dt">condition =</span> condition_exp, 
  <span class="dt">test =</span> <span class="st">"lrt"</span>
)
<span class="kw">head</span>(res_exp)</code></pre></div>
<pre><code>##      id log2.fold         pval mean_LumA mean_other     mean
## 1 AURKB  2.339920 3.191000e-32  539.0426  2323.8868  1485.01
## 2  CBX2  2.895062 2.834335e-26  632.5106  4296.6038  2574.48
## 3 KPNA2  1.447288 8.551812e-24  11547.36   26427.38 19433.77
## 4 PRR11  3.822148 2.286874e-22   396.383   3479.981  2030.69
## 5 BIRC5  1.988998 1.953941e-21  1957.085   6658.358  4448.76
## 6  GSG2  1.405039 3.527773e-21  278.2128   629.3396   464.31</code></pre>
<p>As a result we get a data frame with columns describing the log2 of the fold change, p-value and the mean for each gene.</p>
</div>
</div>
<div id="identification-of-dmr---differentially-methylated-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#identification-of-dmr---differentially-methylated-regions" class="anchor"></a>Identification of DMR - differentially methylated regions</h2>
<div id="brca_met-data-set" class="section level3">
<h3 class="hasAnchor">
<a href="#brca_met-data-set" class="anchor"></a><code>BRCA_met</code> data set</h3>
<p>In this section, we will work with the methylation level data from TCGA database. Package <code>MLExpRessoData</code> contains <code>BRCA_met</code> dataset. This data set contains information about methylation of CpG probes for patients with breast cancer. Rows of this data set correspond to patients, more precisely, to samples taken from patients. First column <code>SUBTYPE</code> corresponds to a subtype of BRCA cancer, next columns correspond to CpG probes. Values inside the table indicate the percentage methylation level of CpG probe for specified sample.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(BRCA_met)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</code></pre></div>
<pre><code>##                              SUBTYPE cg00021527 cg00031162  cg00032227
## TCGA-A1-A0SD-01A-11D-A112-05    LumA 0.03781858  0.7910348 0.006391233
## TCGA-A2-A04N-01A-11D-A112-05    LumA 0.01437552  0.7359370 0.008752293
## TCGA-A2-A04P-01A-31D-A032-05   Basal 0.01360124  0.6967802 0.009442039
## TCGA-A2-A04Q-01A-21D-A032-05   Basal 0.01525656  0.5341244 0.014674247
## TCGA-A2-A04T-01A-21D-A032-05   Basal 0.01167384  0.7378100 0.012251559</code></pre>
</div>
<div id="data-preparation" class="section level3">
<h3 class="hasAnchor">
<a href="#data-preparation" class="anchor"></a>Data preparation</h3>
<p>In this analysis we would like to find genes with different methylation. At first we need to use function <code><a href="../reference/aggregate_probes.html">aggregate_probes()</a></code>, which generates new data frame with CpG probes aggregated to genes. To this aggregation we use, by default, the Illumina Human Methylation data set from the <code>TxDb.Hsapiens.UCSC.hg18.knownGene</code> Bioconductor package.</p>
<p>Function <code><a href="../reference/aggregate_probes.html">aggregate_probes()</a></code> requires a parameter <code>data</code> - data frame containing methylation values for CpG probes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">BRCA_met_gen &lt;-<span class="st"> </span><span class="kw"><a href="../reference/aggregate_probes.html">aggregate_probes</a></span>(<span class="dt">data =</span> BRCA_met) 
<span class="kw">head</span>(BRCA_met_gen)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</code></pre></div>
<pre><code>##                                  AANAT    AARSD1       AATF      AATK
## TCGA-A1-A0SD-01A-11D-A112-05 0.7148533 0.8625816 0.24294092 0.7835302
## TCGA-A2-A04N-01A-11D-A112-05 0.5850106 0.8355825 0.21367129 0.8466190
## TCGA-A2-A04P-01A-31D-A032-05 0.4495537 0.8786166 0.03277413 0.3417919
## TCGA-A2-A04Q-01A-21D-A032-05 0.7120650 0.8819490 0.03460160 0.7264985
## TCGA-A2-A04T-01A-21D-A032-05 0.6010397 0.7739978 0.02501599 0.6276399</code></pre>
<p>In our example we will test for differential methylation between groups with <code>LumA</code> breast cancer subtype and other subtypes of that cancer. Again we will use condition vector, which consist of two values corresponds to a subtype of breast cancer: <code>LumA</code> and <code>other</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">condition_met &lt;-<span class="st"> </span><span class="kw">ifelse</span>(BRCA_met<span class="op">$</span>SUBTYPE <span class="op">==</span><span class="st"> "LumA"</span>, <span class="st">"LumA"</span>, <span class="st">"other"</span>)
<span class="kw">head</span>(condition_met, <span class="dv">8</span>)</code></pre></div>
<pre><code>## [1] "LumA"  "LumA"  "other" "other" "other" "other" "LumA"  "other"</code></pre>
</div>
<div id="testing-1" class="section level3">
<h3 class="hasAnchor">
<a href="#testing-1" class="anchor"></a>Testing</h3>
<p>In the <code>MLExpResso</code> package we carry out the tests for identification of differentially methylated regions. To do this we use the <code><a href="../reference/calculate_test.html">calculate_test()</a></code> function.</p>
<p>This function requires parameters such that:</p>
<ul>
<li><p><code>data</code> - object of the class appropriate for the given test. In the example below, we use the <code>BRCA_met_gen</code> methylation data aggregated to probes.</p></li>
<li><p><code>condition</code> - a factor of levels corresponding to order of samples in data. In our example, we use the <code>condition_met</code> vector defined earlier.</p></li>
<li><p><code>test</code> - character defining test. Possible values of parameter <code>test</code> are described in the documentation of this function.</p></li>
</ul>
<p>Methylation tests are based on the methods implemented in packages <code>limma</code> and <code>MethyAnalysis</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_met &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calculate_test.html">calculate_test</a></span>(
  <span class="dt">data =</span> BRCA_met_gen, 
  <span class="dt">condition =</span> condition_met,
  <span class="dt">test =</span> <span class="st">"ttest"</span>
)
<span class="kw">head</span>(res_met)</code></pre></div>
<pre><code>##        id   log2.fold         pval mean_LumA mean_other      mean
## 1   ICAM2 -0.15151320 3.754116e-17 0.2547275  0.4062407 0.3330801
## 2    RILP -0.05073691 2.575168e-13 0.3079069  0.3586438 0.3341447
## 3   PIPOX  0.11505558 5.360053e-12 0.4242804  0.3092248 0.3647812
## 4 TNFSF12 -0.13412855 5.867083e-12 0.1791401  0.3132686 0.2485025
## 5     CD7  0.09822690 1.641919e-11 0.8635077  0.7652808 0.8127112
## 6    KSR1  0.19973400 2.054467e-11  0.658270   0.458536 0.5549808</code></pre>
<p>As a result we get a data frame with columns describing the log2 of the fold change, p-value and the mean for each gene.</p>
</div>
</div>
<div id="identification-regions-with-changes-in-expression-and-methylation" class="section level2">
<h2 class="hasAnchor">
<a href="#identification-regions-with-changes-in-expression-and-methylation" class="anchor"></a>Identification regions with changes in expression and methylation</h2>
<p>We can also create a comparison table with results of <code><a href="../reference/calculate_test.html">calculate_test()</a></code> function for methylation and expression data.</p>
<p>Function <code><a href="../reference/calculate_comparison_table.html">calculate_comparison_table()</a></code> requires parameters such that:</p>
<ul>
<li><p><code>data1</code>, <code>data2</code> - objects of the class appropriate for the given tests.</p></li>
<li><p><code>condition1</code>, <code>condition2</code> - factors of levels coresponding to order of samples in <code>data1</code> and <code>data2</code> respectively.</p></li>
<li><p><code>test1</code>, <code>test2</code> - characters defining tests coresponding to order of samples in <code>data1</code> and <code>data2</code> respectively. Possible values of parameter <code>test</code> are described in <code><a href="../reference/calculate_test.html">calculate_test()</a></code> function documentation.</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">comparison &lt;-<span class="st"> </span><span class="kw"><a href="../reference/calculate_comparison_table.html">calculate_comparison_table</a></span>(
  <span class="dt">data1 =</span> BRCA_exp[,<span class="op">!</span>(<span class="kw">colnames</span>(BRCA_exp)<span class="op">==</span><span class="st">"SUBTYPE"</span>)], 
  <span class="dt">data2 =</span> BRCA_met_gen, 
  <span class="dt">condition1 =</span> condition_exp, 
  <span class="dt">condition2 =</span> condition_met,
  <span class="dt">test1 =</span> <span class="st">"nbinom2"</span>, 
  <span class="dt">test2 =</span> <span class="st">"ttest"</span>
)
<span class="kw">head</span>(comparison)</code></pre></div>
<pre><code>##        id nbinom2.log2.fold nbinom2.pval ttest.log2.fold   ttest.pval
## 59  AURKB          2.360714 1.704243e-37    0.0017389592 2.077252e-01
## 102  CBX2          2.905397 5.402147e-31    0.0584687549 1.214043e-06
## 327 KPNA2          1.466181 3.396674e-26    0.0012105971 7.505750e-01
## 277  GSG2          1.426569 3.325659e-25   -0.0018566938 2.411495e-01
## 66  BIRC5          2.004989 9.482155e-24   -0.0005444811 5.330216e-01
## 334 KRT16          4.333332 4.102956e-19    0.0486814033 1.606151e-05
##     geom.mean.rank no.probes
## 59    1.881527e-19         2
## 102   8.098418e-19         2
## 327   1.596702e-13         1
## 277   2.831926e-13         2
## 66    2.248153e-12         1
## 334   2.567093e-12         2</code></pre>
<p>As a result, we get a data frame with columns describing the log2 of the fold change, p-value and the mean for each gene for two tests. With this two test results, we compute the ranking of the most significant changed genes in terms of both methylation and expression. The created column contains the geometric mean of p-values for expression and methylation.</p>
</div>
<div id="visualization-of-identified-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization-of-identified-regions" class="anchor"></a>Visualization of identified regions</h2>
<p>The great advantage of <code>MLExpResso</code> package is the ability to perform a variety of visualizations for expression and methylation. All plots in our package are based on the <code>ggplot2</code> package. We use also the <code>scales</code> and <code>ggrepel</code> packages for mathematical axes and repel overlapping text labels.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">test_exp &lt;-<span class="st"> </span>comparison[ , <span class="kw">c</span>(<span class="st">"id"</span>, <span class="st">"nbinom2.log2.fold"</span>, <span class="st">"nbinom2.pval"</span>)]
test_met &lt;-<span class="st"> </span>comparison[ , <span class="kw">c</span>(<span class="st">"id"</span>, <span class="st">"ttest.log2.fold"</span>, <span class="st">"ttest.pval"</span>)]</code></pre></div>
<p>For both, methylation and expression data, we can visualise the volcano plots for results of chosen tests and simple statistics for chosen gene.</p>
<p>Function <code><a href="../reference/plot_volcanoes.html">plot_volcanoes()</a></code> requiers parameters such that:</p>
<ul>
<li><p><code>data.m</code>, <code>data.e</code> - data sets with information from methylation and expression respectively. In the example below, we use the <code>BRCA_met</code> and <code>BRCA_exp</code> data without the <code>SUBTYPE</code> columns.</p></li>
<li><p><code>condition.m</code>, <code>condition.e</code> - factors of levels coresponding to the order of samples in <code>data.m</code> and <code>data.e</code> respectively. In our example, we use the <code>condition_met</code> and <code>condition_exp</code> vectors defined earlier.</p></li>
<li><p><code>gene</code> - character defining the gene name. In the example, we visualize results for <code>CACNA1G</code> gene</p></li>
<li><p><code>test.m</code>, <code>test.e</code> - results from <code><a href="../reference/calculate_test.html">calculate_test()</a></code> function for methylation an expression data.</p></li>
<li><p><code>values</code> - if <code>TRUE</code> p-values and log fold for the chosen gene will be added to the plot.</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plot_volcanoes.html">plot_volcanoes</a></span>(
  <span class="dt">data.m =</span> BRCA_met[,<span class="op">!</span>(<span class="kw">colnames</span>(BRCA_met) <span class="op">==</span><span class="st"> "SUBTYPE"</span>)],
  <span class="dt">data.e =</span> BRCA_exp[,<span class="op">!</span>(<span class="kw">colnames</span>(BRCA_exp) <span class="op">==</span><span class="st"> "SUBTYPE"</span>)],
  <span class="dt">condition.m =</span> condition_met, 
  <span class="dt">condition.e =</span> condition_exp, 
  <span class="dt">gene =</span> <span class="st">"CACNA1G"</span>, 
  <span class="dt">test.m =</span> test_met, 
  <span class="dt">test.e =</span> test_exp,
  <span class="dt">values=</span><span class="ot">TRUE</span>
)</code></pre></div>
<p><img src="QuickStart_files/figure-html/unnamed-chunk-9-1.png" width="1824"></p>
<p>Other function <code><a href="../reference/plot_gene.html">plot_gene()</a></code> allow us to visualize the <code>methylation path</code> - placement of probes near the gene with a marked percentage of methylation for each probe in division into groups.</p>
<p>Function <code><a href="../reference/plot_gene.html">plot_gene()</a></code> requiers parameters such that:</p>
<ul>
<li><p><code>data.m</code>, <code>data.e</code> - data sets with informations from methylation and expression respectively. Note that <code><a href="../reference/plot_gene.html">plot_gene()</a></code> methylation requires data frame with CpG probes, not genes. In the example below, we use the <code>BRCA_met</code> and <code>BRCA_exp</code>.</p></li>
<li><p><code>condition.m</code>, <code>condition.e</code> - factors of levels coresponding to order of samples in <code>data.m</code> and <code>data.e</code> respectively. In our example, we use the <code>condition_met</code> and <code>condition_exp</code> vectors defined earlier.</p></li>
<li><p><code>gene</code> - character defining the gene name. In the example, we visualize results for <code>CACNA1G</code> gene.</p></li>
<li><p>… additional parameters. Below we use parameters <code>show.gene</code> - If <code>TRUE</code> line corresponding to the gene will be drawn, <code>observ</code> - If <code>TRUE</code> dots corresponding to CpG probes will be drawn and <code>islands</code> -If <code>TRUE</code> line corresponding to islands should be drawn.</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plot_gene.html">plot_gene</a></span>(
  <span class="dt">data.m =</span> BRCA_met, 
  <span class="dt">data.e =</span> BRCA_exp, 
  <span class="dt">condition.m =</span> condition_met, 
  <span class="dt">condition.e =</span> condition_exp, 
  <span class="dt">gene =</span> <span class="st">"CACNA1G"</span>,
  <span class="dt">show.gene =</span> <span class="ot">TRUE</span>,
  <span class="dt">observ =</span> <span class="ot">TRUE</span>,
  <span class="dt">islands =</span> <span class="ot">TRUE</span>
)</code></pre></div>
<p><img src="QuickStart_files/figure-html/unnamed-chunk-10-1.png" width="1824"></p>
<p>Using this function we also get boxplots containing values of expression in division from <code>condition_exp</code> vector for chosen gene.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      true
    </div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aleksandra Dabrowska, Alicja Gosiewska, Przemyslaw Biecek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
