<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Statistical computations for methylation and expression data. — calculate_test â€¢ MLExpResso</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">MLExpResso</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QuickStart.html">MLExpResso: differential expression and methylation analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Statistical computations for methylation and expression data.</h1>
    </div>

    
    <p>Function <code>calculate_test</code> computes log folds and p-values for choosen test for data from methylation or expression.
The function uses: t-test, negative binomial test, likelihood-ratio test(LRT), quasi-likelihood F-test(QLF).
By default function calls the <code>t-test</code>.</p>
    

    <pre class="usage"><span class='fu'>calculate_test</span>(<span class='no'>data</span>, <span class='no'>condition</span>, <span class='kw'>test</span> <span class='kw'>=</span> <span class='st'>"ttest"</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>An object of the class appropriate for the given test. More in <code>details</code> section.</p></td>
    </tr>
    <tr>
      <th>condition</th>
      <td><p>Factor of levels coresponding to order of samples in data.</p></td>
    </tr>
    <tr>
      <th>test</th>
      <td><p>Variable defining test. Values: ttest, nbinom, nbinom2, lrt, qlf, methyanalysis. More in <code>details</code> section.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Other parameters e.g. `adjust.method` for argument `ttest`.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data frame with the following columns:</p>
<dt>id</dt><dd><p>The id of the observable, taken from the row names of the counts slots.</p></dd>
 <dt>log2.fold</dt><dd><p>The log2 of the fold change.</p></dd>
 <dt>pval</dt><dd><p>The p-values for rejecting the null hypothesis about the means equality.</p></dd>
 <dt>mean</dt><dd><p>Column correspond to means for each gene defined by condition and mean for all probes.</p></dd>

    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Each test may require different data. In this section we will describe details for each availible test:</p><dl class='dl-horizontal'>
  <dt>ttest</dt><dd><p>Student's t-test 
  Test for expression and methylation.
  This test is based on function <code><a href='http://www.rdocumentation.org/packages/limma/topics/lmFit'>lmFit</a></code> from <code>limma</code> package. 
  For this test you should aggregate CpG probes to genes using function: <code><a href='aggregate_probes.html'>aggregate_probes</a></code>.
  Data for this test should have following structure: columns correspond to genes, rows to samples.</p></dd>
  <dt>nbinom</dt><dd><p>Negative binomial test
  Test for expression.
  This test is based on function <code><a href='http://www.rdocumentation.org/packages/DESeq/topics/nbinomTest'>nbinomTest</a></code> from <code>DESeq</code> package. 
  Data for this test should have following structure: columns correspond to genes, rows to samples. 
  Calculations may take some time. It is suggested to use <code>nbinom2</code> parameter.</p></dd>
  <dt>nbinom2</dt><dd><p>Negative binomial test
  Test for expression.
  This test is based on function <code><a href='http://www.rdocumentation.org/packages/DESeq2/topics/DESeq'>DESeq</a></code> from <code>DESeq2</code> package. 
  Data for this test should have following structure: columns correspond to genes, rows to samples.</p></dd>
  <dt>lrt</dt><dd><p>Likelihood-ratio test (LRT)
  Test for expression.
  This test is based on function <code><a href='http://www.rdocumentation.org/packages/edgeR/topics/glmfit'>glmFit</a></code> and <code><a href='http://www.rdocumentation.org/packages/edgeR/topics/glmfit'>glmLRT</a></code> from <code>edgeR</code> package. 
  Data for this test should have following structure: columns correspond to genes, rows to samples.</p></dd>
  <dt>qlf</dt><dd><p>Quasi-likelihood F-test (QLF)
  Test for expression.
  This test is based on functions <code><a href='http://www.rdocumentation.org/packages/edgeR/topics/glmQLFTest'>glmQLFit</a></code> and <code><a href='http://www.rdocumentation.org/packages/edgeR/topics/glmQLFTest'>glmQLFTest</a></code> from <code>edgeR</code> package. 
  Data for this test should have following structure: columns correspond to genes, rows to samples.</p></dd>
  <dt>methyanalysis</dt><dd><p>Slide window smoothing
  Test for methylation.
  This test is based on function <code><a href='http://www.rdocumentation.org/packages/methyAnalysis/topics/detectDMR.slideWin'>detectDMR.slideWin</a></code> from <code>methyAnalysis</code> package.
  It requires a special class of argument data - <code>MethyGenoSet</code>.</p></dd>
</dl>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <p><code><a href='calculate_comparison_table.html'>calculate_comparison_table</a>, <a href='aggregate_probes.html'>aggregate_probes</a></code></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'># NOT RUN {</span>
<span class='fu'>library</span>(<span class='no'>MLExpRessoData</span>)
<span class='no'>BRCA_methylation_gene</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='aggregate_probes.html'>aggregate_probes</a></span>(<span class='no'>BRCA_methylation_all</span>)

<span class='no'>condition_m</span> <span class='kw'>&lt;-</span> <span class='fu'>ifelse</span>(<span class='no'>BRCA_methylation_all</span>$<span class='no'>SUBTYPE</span> <span class='kw'>==</span> <span class='st'>"LumA"</span>, <span class='st'>"LumA"</span>, <span class='st'>"other"</span>)
<span class='no'>test_methylation</span> <span class='kw'>&lt;-</span> <span class='fu'>calculate_test</span>(<span class='no'>BRCA_methylation_gene</span>, <span class='no'>condition_m</span>, <span class='st'>"ttest"</span>)

<span class='co'># }</span><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Aleksandra Dabrowska, Alicja Gosiewska, Przemyslaw Biecek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
